Vagrantfile

```diff
@@ -0,0 +1,146 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.require_version ">= 1.5"
+require "yaml"
+
+def testing?
+  !ENV["ASHLAR_ENV"].nil? && ENV["ASHLAR_ENV"] == "TEST"
+end
+
+# Deserialize Ansible Galaxy installation metadata for a role
+def galaxy_install_info(role_name)
+  role_path = File.join("deployment", "ansible", "roles", role_name)
+  galaxy_install_info = File.join(role_path, "meta", ".galaxy_install_info")
+
+  if (File.directory?(role_path) || File.symlink?(role_path)) && File.exists?(galaxy_install_info)
+    YAML.load_file(galaxy_install_info)
+  else
+    { install_date: "", version: "0.0.0" }
+  end
+end
+
+# Uses the contents of roles.txt to ensure that ansible-galaxy is run
+# if any dependencies are missing
+def install_dependent_roles
+  ansible_directory = File.join("deployment", "ansible")
+  ansible_roles_txt = File.join(ansible_directory, "roles.txt")
+
+  File.foreach(ansible_roles_txt) do |line|
+    role_name, role_version = line.split(",")
+    role_path = File.join(ansible_directory, "roles", role_name)
+    galaxy_metadata = galaxy_install_info(role_name)
+
+    if galaxy_metadata["version"] != role_version.strip
+      unless system("ansible-galaxy install -f -r #{ansible_roles_txt} -p #{File.dirname(role_path)}")
+        $stderr.puts "\nERROR: An attempt to install Ansible role dependencies failed."
+        exit(1)
+      end
+
+      break
+    end
+  end
+end
+
+# Install missing role dependencies based on the contents of roles.txt
+if [ "up", "provision" ].include?(ARGV.first)
+  install_dependent_roles
+end
+
+if !ENV["VAGRANT_ENV"].nil? && ENV["VAGRANT_ENV"] == "TEST"
+  ANSIBLE_ENV_GROUPS = {
+    "test:children" => [
+      "app-servers",
+      "database-servers"
+    ]
+  }
+  VAGRANT_NETWORK_OPTIONS = { auto_correct: true }
+else
+  ANSIBLE_ENV_GROUPS = {
+    "development:children" => [
+      "app-servers",
+      "database-servers"
+    ]
+  }
+  VAGRANT_NETWORK_OPTIONS = { auto_correct: false }
+end
+
+ANSIBLE_GROUPS = {
+  "app-servers" => [ "app" ],
+  "database-servers" => [ "database" ]
+}
+
+VAGRANT_PROXYCONF_ENDPOINT = ENV["VAGRANT_PROXYCONF_ENDPOINT"]
+VAGRANTFILE_API_VERSION = "2"
+
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.box = "ubuntu/trusty64"
+  config.hostmanager.enabled = true
+  config.hostmanager.manage_host = false
+  config.hostmanager.ignore_private_ip = false
+  config.hostmanager.include_offline = true
+
+  # Wire up the proxy if:
+  #
+  #   - The vagrant-proxyconf Vagrant plugin is installed
+  #   - The user set the VAGRANT_PROXYCONF_ENDPOINT environmental variable
+  #
+  if Vagrant.has_plugin?("vagrant-proxyconf") &&
```

Comment:
Because this never plays well with Java dependency managers, it may not be worth keeping. Currently experimenting with `vagrant-cachier` in the Model My Watershed project:

```ruby
# Wire up package caching:
if Vagrant.has_plugin?("vagrant-cachier")
  config.cache.scope = :machine
end
```

It using synced folders to map into the directories where packages get downloaded to. So, ultimately, the packages are being written to the virtual machine host and mapped into the right spots on the virtual machine.